op-program/l2/engineapi/l2_engine_api.go

```diff
@@ -102,7 +92,7 @@ func computePayloadId(headBlockHash common.Hash, params *eth.PayloadAttributes)
 }
 
 func (ea *L2EngineAPI) RemainingBlockGas() uint64 {
-	return ea.l2GasPool.Gas()
+	return ea.blockProcessor.gasPool.Gas()
```

Comment:
Maybe return 0 if the `blockProcessor` is nil?

Reply:
Yep makes sense.